WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -3949,6 +3955,21 @@ public void onEventMainThread(UploadService.UploadMediaRetryEvent event) {
         }
     }
 
+    private boolean shouldShowAsyncPromoDialog() {
+        // To make sure the behavior matches what we're communicating to the user as available options,
+        // the same conditions as per `getSaveButtonText()` apply:
+        //  if status is DRAFT and isNewPost() && mPost.isLocalDraft() --> PUBLISH;
+        //  else if UNKNOWN and mPost.isLocalDraft() --> PUBLISH
+        //  else if SAVE/UPDATE
+        PostStatus status = PostStatus.fromPost(mPost);
+        if (AppPrefs.isAsyncPromoRequired() && userCanPublishPosts()
```

Comment:
Nicely spotted! thanks for asking the question - we also check for `PUBLISHED` posts here because the user can have a new post (`isNewPost()` is `true`) and they may edit the Post settings and change the status to `Publish` (last of the "test case A" in the PR description): they still can `Save as a draft` (which will bring it back to a `draft` status), but tapping on the action bar's main action would upload the Post to the server and it'd be saved with the `published` status as per the user action so, in that case, we also need to show the async promo dialog (if the other conditions apply like, it was not shown before, etc).

Reply:
added a comment in code here 4a47352d91 👍

Reply:
Thanks;)